### ISSUE TITLE
JSON Decode Error

### ISSUE THREAD

**Issue by Brewski63** [UMNFuN/FUNGuild#41]
I apologise if this is a naive question but I haven't been able to find any similar answers online.

I ran FUNGuild on some data around October 2020 and it worked no problem, but our data has since changed and I need to rerun everything. I tried to rerun my old data using everything the exact same as I did in October but I am coming up with the following error:

C:\Users . . . FUNGuild-master>python Guilds_v1.1.py -otu esv_table_funguild.txt -db fungi 

Guilds_v1.1.py:92: DeprecationWarning: 'U' mode is deprecated
  with open(otu_file, 'rU') as f1:
FunGuild v1.0 Beta
Connecting with FUNGuild database ...
Traceback (most recent call last):
  File "Guilds_v1.1.py", line 128, in <module>
    current_record = json.loads(record)
  File "C:\Users\BrewDaddy\anaconda3\lib\json\__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "C:\Users\BrewDaddy\anaconda3\lib\json\decoder.py", line 340, in decode
    raise JSONDecodeError("Extra data", s, end)
json.decoder.JSONDecodeError: Extra data: line 1 column 436 (char 435)

I found one forum that suggested replacing '%s' with '%s\n' to: "output.write('%s\n' % ('\t'.join(header))) #Header", "output.write('%s\n' % rec)" and "output_unmatched.write('%s\n' % rec)" (lines 290, 296 and 326) but that came up with the same error. I also read that it could be fixed by using Anaconda3 but that is what I currently use. Just curious to know what I am doing wrong for it to not work this time around.

Kind Regards,
Brewski63

**Comment by ZeweiSong** [comment]
Can you make sure that you are using the latest script? We currently make some slight change to it.

Cheers,

Zewei

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by Brewski63** [comment]
Dear Zewei,

Thank you so much for your prompt response. I was in fact using the older version the script, and by using the latest version I was able to resolve that error. However, I encountered another error thereafter:

(base) C:\ . . . >python Guilds_v1.1.py -otu esv_table_funguild.txt -db fungi
FunGuild v1.1 Beta
Connecting with FUNGuild database ...

Reading in the OTU table: 'esv_table_funguild.txt'

Searching the FUNGuild database...
10%
20%
30%
40%
50%
60%
70%
80%
90%
100%

Found 1146 matching taxonomy records in the database.
Dereplicating and sorting the result...
Traceback (most recent call last):
  File "Guilds_v1.1.py", line 355, in <module>
    f.write('%s\n' % output_line)
  File "C:\Users\BrewDaddy\anaconda3\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u0142' in position 542: character maps to <undefined>

I managed to solve this error by adding " , encoding='utf-8' " to the open function in line 343 of your script. I have since successfully ran the funguild program and have obtained my data. Thank you for your assistance.

Kind Regards,
Matthew Brewer.
